File: aws_api/SermonsAPI/Models/YouTubeModels.cs
using System;
using System.Collections.Generic;

namespace GlobalArticleDatabaseAPI.Models
{
  public class GetItemsResponse
  {
    public string kind { get; set; }
    public string etag { get; set; }
    public string nextPageToken { get; set; }
    public string regionCode { get; set; }
    public PageInfo pageInfo { get; set; }
    public List<Item> items { get; set; }
  }

  public class PageInfo
  {
    public int totalResults { get; set; }
    public int resultsPerPage { get; set; }
  }

  public class Item
  {
    public string kind { get; set; }
    public string etag { get; set; }
    public dynamic id { get; set; }
    public Snippet snippet { get; set; }
  }

  public class ItemId
  {
    public string kind { get; set; }
    public string videoId { get; set; }
  }

  public class Snippet
  {
    public DateTime publishedAt { get; set; }

    public string channelId { get; set; }
    public string title { get; set; }
    public string description { get; set; }
    public Thumbnails thumbnails { get; set; }
    public string channelTitle { get; set; }
    public string liveBroadcastContent { get; set; }
    public DateTime Date { get; set; }
    public string playlistId { get; set; }
    public int position { get; set; }
    public dynamic resourceId { get; set; }
  }

  public class Thumbnails
  {
    public Thumbnail @default { get; set; }
    public Thumbnail medium { get; set; }
    public Thumbnail high { get; set; }
  }

  public class Thumbnail
  {
    public string url { get; set; }
    public int width { get; set; }
    public int height { get; set; }
  }
}

File: src/app/models/youtube.models.ts
export class GetPlayListResponse {
  kind: string;
  etag: string;
  nextPageToken: string;
  regionCode: string;
  pageInfo: PageInfo;
  items: Item[];
}

export class PageInfo {    
    totalResults: number;
    resultsPerPage: number;      
}

export class Item {
    kind: string;
    etag: string;
    id: any;    
    snippet: Snippet;

}

export class ItemId {
    kind: string;
    videoId: string;
}

export class Snippet {
    publishedAt: Date;
    channelId: string;
    title: string;
    description: string;
    thumbnails: Thumbnails;
    channelTitle: string;
    liveBroadcastContent: string;
    publishTime: Date;
    playlistId: string;
    position: number;
    resourceId: ItemId;
}

export class Thumbnails {
    default: Thumbnail;
    medium: Thumbnail;
    high: Thumbnail;
}

export class Thumbnail {
    url: string;
    width: number;
    height: number;
}